File: Assets/Game/Script/Framework/Net/CsProtocol/Protocol/Login/GetPlayerInfoResponse.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Spring.Utils;

namespace ZJYFrameWork.Net.CsProtocol.Buffer
{
    public class GetPlayerInfoResponse: IPacket
    {
        public static GetPlayerInfoResponse ValueOf()
        {
            var packet = new GetPlayerInfoResponse
            {
            };
            return packet;
        }
        public short ProtocolId()
        {
            return 1005;
        }
    }
    public class GetPlayerInfoResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1005;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                buffer.WriteBool(false);
                return;
            }
            var InfoRequest = (GetPlayerInfoResponse)packet;
            var message = new ServerMessageWrite(InfoRequest.ProtocolId(), InfoRequest);
            var json = JsonConvert.SerializeObject(message);
            Debug.Log(json);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            var json = StringUtils.BytesToString(buffer.ToBytes());
            var dict = JsonConvert.DeserializeObject<Dictionary<object, object>>(json);
            var packet = JsonConvert.DeserializeObject<GetPlayerInfoResponse>(dict["packet"].ToString());
            return packet;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/login/getPlayerInfoResponse.lua
---@class GetPlayerInfoResponse
local GetPlayerInfoResponse = class("GetPlayerInfoResponse")
function GetPlayerInfoResponse:ctor()
    
end


---@return GetPlayerInfoResponse
function GetPlayerInfoResponse:new()
    
    return self
end

---@return number
function GetPlayerInfoResponse:protocolId()
    return 1012
end

---@return string
function GetPlayerInfoResponse:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return GetPlayerInfoResponse
function GetPlayerInfoResponse:read(data)

    local packet = self:new(
            )
    return packet
end



return GetPlayerInfoResponse
